fdbclient/BackupContainerFileSystem.actor.cpp

```diff
@@ -1133,16 +1133,17 @@ class BackupContainerFileSystemImpl {
 		    filename,
 		    IAsyncFile::OPEN_ATOMIC_WRITE_AND_CREATE | IAsyncFile::OPEN_READWRITE | IAsyncFile::OPEN_CREATE,
 		    0600));
-		StreamCipher::Key::RawKeyType testKey;
-		generateRandomData(testKey.data(), testKey.size());
+		StreamCipherKey testKey(AES_256_KEY_LENGTH);
+		testKey.initializeRandomTestKey();
+		// generateRandomData(testKey.data(), testKey.size());
 		keyFile->write(testKey.data(), testKey.size(), 0);
 		wait(keyFile->sync());
 		return Void();
 	}
 
 	ACTOR static Future<Void> readEncryptionKey(std::string encryptionKeyFileName) {
 		state Reference<IAsyncFile> keyFile;
-		state StreamCipher::Key::RawKeyType key;
+		state StreamCipherKey* cipherKey = StreamCipherKey::getGlobalCipherKey();
```

Comment:
IMO `getGlobalCipherKey` should return a `StreamCipherKey const *`. We can use some other static method in `StreamCipherKey` to set the global key.

Reply:
Fair point. I plan to introduce "StreamCipher::setGlobalKey()" method that can be called from "StreamCipherKey::getGlobalKey" as needed.